aeon/anomaly_detection/distribution_based/_idk.py

```diff
@@ -0,0 +1,236 @@
+        width: int = 1,
+        t: int = 100,
+        sliding: bool = False,
+        random_state: Optional[int] = None,
+    ) -> None:
+        self.psi1 = psi1
+        self.psi2 = psi2
+        self.width = width
+        self.t = t
+        self.sliding = sliding
+        self.random_state = random_state
+        super().__init__(axis=0)
+
+    def _compute_point_to_sample(self, X, sample_indices):
+        sample = X[sample_indices, :]
+        tem1 = np.dot(np.square(X), np.ones(sample.T.shape))
+        tem2 = np.dot(np.ones(X.shape), np.square(sample.T))
+        point2sample = tem1 + tem2 - 2 * np.dot(X, sample.T)
+
+        sample2sample = point2sample[sample_indices, :]
+        row, col = np.diag_indices_from(sample2sample)
+        sample2sample[row, col] = np.nan
+
+        radius_list = np.nanmin(sample2sample, axis=1)
+        min_dist_point2sample = np.argmin(point2sample, axis=1)
+
+        return point2sample, radius_list, min_dist_point2sample
+
+    def _generate_feature_map(self, X, psi, t, rng):
+        feature_matrix = np.zeros((X.shape[0], t * psi), dtype=int)
+        for time in range(t):
+            sample_indices = rng.choice(len(X), size=psi, replace=False)
+            point2sample, radius_list, min_dist_point2sample = (
+                self._compute_point_to_sample(X, sample_indices)
+            )
+
+            min_point2sample_index = np.argmin(point2sample, axis=1)
+            min_dist_point2sample = min_point2sample_index + time * psi
+            point2sample_value = point2sample[
+                range(len(feature_matrix)), min_point2sample_index
+            ]
+            ind = point2sample_value < radius_list[min_point2sample_index]
+            feature_matrix[ind, min_dist_point2sample[ind]] = 1
+
+        return feature_matrix
+
+    def _compute_idk_score(self, X, psi, t, rng):
+        point_fm_list = self._generate_feature_map(X=X, psi=psi, t=t, rng=rng)
+        feature_mean_map = np.mean(point_fm_list, axis=0)
+        return np.dot(point_fm_list, feature_mean_map) / t
+
+    def _idk_fixed_window(self, X, rng):
+        window_num = int(np.ceil(X.shape[0] / self.width))
+        featuremap_count = np.zeros((window_num, self.t * self.psi1))
+        onepoint_matrix = np.full((X.shape[0], self.t), -1)
+
+        for time in range(self.t):
+            sample_indices = rng.choice(X.shape[0], size=self.psi1, replace=False)
+            point2sample, radius_list, min_dist_point2sample = (
+                self._compute_point_to_sample(X, sample_indices)
+            )
+
+            for i in range(X.shape[0]):
+                if (
+                    point2sample[i][min_dist_point2sample[i]]
+                    < radius_list[min_dist_point2sample[i]]
+                ):
+                    onepoint_matrix[i][time] = (
+                        min_dist_point2sample[i] + time * self.psi1
+                    )
+                    featuremap_count[(int)(i / self.width)][
```

Comment:
```suggestion
                    featuremap_count[int(i / self.width)][
```
looks a bit cleaner (also in other places)

Reply:
Fixed